Feature/iwyu part 4

petersteneteg opened the pull request:
Changes proposed in this PR:
 * Include cleanup for a number of modules

This should be the last of the core modules

petersteneteg committed:
QtWidgets: iwyu

petersteneteg committed:
QtWidgets: remove qt blanket warning ignores

petersteneteg committed:
Tools: iwyu Qt update

petersteneteg committed:
QtWidget: iwyu update

petersteneteg committed:
DataFrameQt: iwyu

petersteneteg committed:
AnimationQt: iwyu

petersteneteg committed:
Python3Qt: iwyu

petersteneteg committed:
OpenGLQt: iwyu

petersteneteg committed:
AnimationQt: remove qt blanket warning ignores

petersteneteg committed:
Python3Qt: incfix

Inviwo-Team committed:
Jenkins: Format fixes

martinfalk commented:
Include removal of `QMenu` was a bit too aggressive.
```
Error: D:\a\inviwo\inviwo\inviwo\modules\qtwidgets\include\modules/qtwidgets/properties/anglepropertywidgetqt.h(117,33): error C2027: use of undefined type 'QMenu' [D:\a\inviwo\inviwo\build\modules\qtwidgets\inviwo-module-qtwidgets.vcxproj]
```

```
Warning: /home/runner/work/inviwo/inviwo/inviwo/modules/qtwidgets/include/modules/qtwidgets/properties/ordinalminmaxpropertywidgetqt.h:291:31: warning: invalid use of incomplete type ‘class QMenu’
  291 |     auto settingsAction = menu->addAction(tr("&Property settings..."));
      |                               ^~
In file included from /home/runner/work/inviwo/inviwo/inviwo/modules/qtwidgets/include/modules/qtwidgets/properties/ordinalminmaxpropertywidgetqt.h:37,
                 from /home/runner/work/inviwo/inviwo/inviwo/modules/qtwidgets/src/properties/ordinalminmaxpropertywidgetqt.cpp:30:
/home/runner/work/inviwo/inviwo/inviwo/modules/qtwidgets/include/modules/qtwidgets/inviwoqtutils.h:66:7: note: forward declaration of ‘class QMenu’
   66 | class QMenu;
      |       ^~~~~
```

petersteneteg committed:
AnimationQt: incfix

petersteneteg committed:
QtWidget: incfix

petersteneteg committed:
OpenGL: incfix

msvc will instanciate DataInport<Volume, 1, false> in one of the function delcs, so we need the definitions

Inviwo-Team committed:
Jenkins: Format fixes

petersteneteg committed:
QTWidget: incfix

ResearchDaniel commented on modules/animationqt/src/animationviewqt.cpp:
```diff
@@ -28,22 +28,44 @@
+
+#include <cstdlib>  // for abs, size_t
+#include <cmath>    // for floor, abs
+
```
Add back
include <warn/push>
 #include <warn/ignore/all>

ResearchDaniel commented on modules/animationqt/src/demo/demonavigatordockwidgetqt.cpp:
```diff
@@ -27,31 +27,37 @@
+#include <memory>  // for unique_ptr
+#include <string>  // for string
+#include <vector>  // for vector
+
```
Warn push/pop

ResearchDaniel commented on modules/animationqt/src/sequenceeditor/controlsequenceeditor.cpp:
```diff
@@ -29,25 +29,26 @@
+#include <modules/qtwidgets/inviwoqtutils.h>                          // for toQString
+
+#include <cstddef>  // for size_t
+
```
Warn push/pop

ResearchDaniel commented on modules/animationqt/src/sequenceeditor/keyframeeditorwidget.cpp:
```diff
@@ -28,25 +28,24 @@
+
+#include <functional>  // for __base
+
+#include <QComboBox>       // for QComboBox
```
Warn push/pop

ResearchDaniel commented on modules/animationqt/src/sequenceeditor/propertysequenceeditor.cpp:
```diff
@@ -28,31 +28,49 @@
+#include <memory>            // for unique_ptr, unique...
+#include <vector>            // for vector
+
+#include <QComboBox>       // for QComboBox
```
Warn push/pop

ResearchDaniel commented on modules/animationqt/src/sequenceeditor/sequenceeditorwidget.cpp:
```diff
@@ -29,24 +29,22 @@
-#include <QLabel>
-#include <QComboBox>
-#include <warn/pop>
+#include <QVBoxLayout>  // for QVBoxLayout
```
Warn push/pop?

ResearchDaniel commented on modules/animationqt/src/trackcontrolswidgetqt.cpp:
```diff
@@ -28,17 +28,29 @@
+#include <iterator>   // for prev
+#include <vector>     // for vector
+
+#include <QAction>       // for QAction
```
Warn push/pop?

ResearchDaniel commented on modules/animationqt/src/widgets/editorconstants.cpp:
```diff
@@ -29,14 +29,15 @@
+#include <cmath>      // for round, remainder, pow, abs, floor, log10
+#include <limits>     // for numeric_limits
+
+#include <QApplication>  // for QApplication
```
Warn push/pop

ResearchDaniel commented on modules/animationqt/src/widgets/keyframesequencewidgetqt.cpp:
```diff
@@ -29,22 +29,38 @@
+#include <cstddef>    // for size_t
+#include <utility>    // for pair
+
+#include <QApplication>     // for QApplication
```
Warn push/pop

ResearchDaniel commented on modules/animationqt/src/animationeditordockwidgetqt.cpp:
```diff
@@ -29,51 +29,71 @@
-#include <QMenuBar>
-#include <QWidget>
-#include <QScrollBar>
-#include <warn/pop>
```
Warn pop removed, but not push

petersteneteg replied:
good catch

ResearchDaniel reviewed (changes requested):
Is it intentional that warn/push/pop is removed in many places?

petersteneteg commented:
yes, the warn pop/push are removed intentionally. It seems most most warnings they were needed for has been fixed upstream

ResearchDaniel commented:
> yes, the warn pop/push are removed intentionally. It seems most most warnings they were needed for has been fixed upstream

Ok. I saw at least one place where the pop was removed, but not the push

petersteneteg committed:
AnimationQt: incfix

ResearchDaniel commented on tools/iwyu/README.md:
```diff
@@ -32,6 +32,30 @@ Example preset to for running iwyu
 
-
-
+Fixes for qt, modify iwyu to avoid some circular include in global.h see
```
```suggestion
Fixes for qt. Modify IWYU (Include What You Use) to avoid some circular include in global.h, see
```

ResearchDaniel commented:
@petersteneteg The following includes are incompatible with Qt5:
```
#include <QEventPoint>      // for QEventPoint, QEventPoi...
#include <QInputDevice>     // for QInputDevice, QInputDe...
#include <QPointingDevice>  // for QPointingDevice
#include <QIODeviceBase>    // for QIODeviceBase::WriteOnly
```

petersteneteg commented:
Ahh... I forgot about qt5 when doing this. Are you building with qt5 still? I just updated to 6.4, is there any good reason to keep qt5 support?

ResearchDaniel commented:
Using it on one of my computers still. No special reason except for backwardscompatibility I think.

petersteneteg commented:
Do you think we should keep supporting it?

ResearchDaniel commented:
How about bumping version and loosing support?
